src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/DefaultJsonTypeInfoResolver.cs

```diff
@@ -84,11 +85,16 @@ private JsonTypeInfo CreateJsonTypeInfo(Type type, JsonSerializerOptions options
 #endif
         }
 
+        [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
+        [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
+        private static JsonTypeInfo<T> CreateReflectionJsonTypeInfo<T>(JsonSerializerOptions options) => new ReflectionJsonTypeInfo<T>(options);
+
         /// <summary>
         /// List of JsonTypeInfo modifiers. Modifying callbacks are called consecutively after initial resolution
         /// and cannot be changed after GetTypeInfo is called.
         /// </summary>
-        public IList<Action<JsonTypeInfo>> Modifiers { get; }
+        public IList<Action<JsonTypeInfo>> Modifiers => _modifiers ??= new ModifierCollection(this);
```

Comment:
if we make this lazy initialize we should at least use `_modifiers` in the GetTypeInfo so it won't instantiate on first usage

Reply:
Yeah, good point.